src/Aspire.Dashboard/Mcp/ResourceMcpTools.cs

```diff
@@ -0,0 +1,385 @@
+        else
+        {
+            return $"Unable to find a resource named '{resourceName}'.";
+        }
+
+        var logParser = new LogParser(ConsoleColor.Black);
+        var logEntries = new LogEntries(maximumEntryCount: AIHelpers.ConsoleLogsLimit) { BaseLineNumber = 1 };
+
+        // Add a timeout for getting all console logs.
+        using var subscribeConsoleLogsCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+        subscribeConsoleLogsCts.CancelAfter(TimeSpan.FromSeconds(20));
+
+        try
+        {
+            await foreach (var entry in _dashboardClient.GetConsoleLogs(resourceName, subscribeConsoleLogsCts.Token).ConfigureAwait(false))
+            {
+                foreach (var logLine in entry)
+                {
+                    logEntries.InsertSorted(logParser.CreateLogEntry(logLine.Content, logLine.IsErrorMessage, resourceName));
+                }
+            }
+        }
+        catch (OperationCanceledException)
+        {
+            return $"Timeout getting console logs for `{resourceName}`";
+        }
+
+        var entries = logEntries.GetEntries().ToList();
+        var totalLogsCount = entries.Count == 0 ? 0 : entries.Last().LineNumber;
+        var (trimmedItems, limitMessage) = AIHelpers.GetLimitFromEndWithSummary<LogEntry>(
+            entries,
+            totalLogsCount,
+            AIHelpers.ConsoleLogsLimit,
+            "console log",
+            AIHelpers.SerializeLogEntry,
+            logEntry => AIHelpers.EstimateTokenCount((string)logEntry));
+        var consoleLogsText = AIHelpers.SerializeConsoleLogs(trimmedItems.Cast<string>().ToList());
+
+        var consoleLogsData = $"""
+            {limitMessage}
+
+            # CONSOLE LOGS
+
+            ```plaintext
+            {consoleLogsText.Trim()}
+            ```
+            """;
+
+        return consoleLogsData;
+    }
+
+    [McpServerTool(Name = "aspire_list_commands"), Description("Lists the command names available for a resource. If a resource needs to be restarted and is currently stopped, use the start command instead.")]
+    public static string GetResourceCommands(IDashboardClient dashboardClient, [Description("The resource name.")] string resourceName)
+    {
+        var resource = dashboardClient.GetResource(resourceName);
+
+        if (resource == null)
+        {
+            throw new McpException($"Resource '{resourceName}' not found.", McpErrorCode.InvalidParams);
+        }
+
+        // Only include commands that can be executed (Enabled).
+
+        return string.Join("\n---\n", resource.Commands.Where(cmd => cmd.State == CommandViewModelState.Enabled).Select(cmd => cmd.Name));
+    }
+
+    [McpServerTool(Name = "aspire_execute_command"), Description("Executes a command on a resource.")]
+    public static async Task ExecuteCommand(IDashboardClient dashboardClient, [Description("The resource name")] string resourceName, [Description("The command name")] string commandName)
```

Comment:
what would happen here if the command uses the interactive service and prompt the user for inputs?

Reply:
@JamesNK This is a great question and something we're going to have to handle. Something to file...